core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -246,14 +255,77 @@ class UserService(
       val platform = CloudPlatform(p)
       val responseProject = if (platform == CloudPlatform.AZURE) {
         val c = AzureManagedAppCoordinates(p.getTenantId, p.getSubscriptionId, p.getManagedResourceGroupId)
-        project.copy(azureManagedAppCoordinates = Some(c))
+        updateLandingZoneStatus(project.copy(azureManagedAppCoordinates = Some(c)))
       } else project
       RawlsBillingProjectResponse(roles, responseProject, platform)
     case (Some(id), None) =>
       val message = Some(s"Unable to find billing profile in Billing Profile Manager for billing profile id: $id")
       RawlsBillingProjectResponse(roles, project.copy(message = message, status = CreationStatuses.Error))
   }
 
+  /**
+    * For billing projects in the status of CreationStatuses.CreatingLandingZone,
+    * retrieve the landing zone status and update the project appropriately.
+    * Deletes the record tracking the landing zone job, if the job is completed.
+    */
+  def updateLandingZoneStatus(billingProject: RawlsBillingProject): RawlsBillingProject = {
+    // if there's no landing zone creation in progress, this is a no-op
+    if (billingProject.status != CreationStatuses.CreatingLandingZone) return billingProject
+    val record = Await.result(
+      workspaceResourceRecordDao.selectByBillingProject(billingProject.projectName).map(_.headOption),
+      DurationInt(30).seconds
```

Comment:
Why the 30 second interval?

Reply:
The actual number arbitrary. 
You have to specify a duration though, and Duration.Inf is probably bad form in real code. 
This is just a db call, so it's generally basically instantaneous, and 30 seconds seemed like enough time to be able to assume there's an issue with the db.

Reply:
Is there a reason this needs to be a synchronous await? If not, it'd be good to retain consistency with the rest of the codebase, where we don't generally await futures outside of tests but rather wrap them in a for comprehension or flatmap.